database/types/pipeline.go

```diff
@@ -193,6 +202,19 @@ func (p *Pipeline) Validate() error {
 		return ErrEmptyPipelineVersion
 	}
 
+	// calculate total size of favorites
+	total := 0
+	for _, w := range p.Warnings {
+		total += len(w)
+	}
+
+	// verify the Favorites field is within the database constraints
+	// len is to factor in number of comma separators included in the database field,
+	// removing 1 due to the last item not having an appended comma
+	if (total + len(p.Warnings) - 1) > constants.PipelineWarningsMaxSize {
+		return ErrExceededWarningsLimit
+	}
```

Comment:
strange way to deal with "favorites" ;) — see code comments

Reply:
:pain_disintegrate: